hollow/src/main/java/com/netflix/hollow/api/producer/validation/ObjectModificationValidator.java

```diff
@@ -0,0 +1,122 @@
+            }
+            BitSet latestOrdinals = typeState.getPopulatedOrdinals();
+            BitSet previousOrdinals = typeState.getPreviousOrdinals();
+            if (previousOrdinals.isEmpty()) {
+                statusBuilder.withMessage("Skipping " + ObjectModificationValidator.class.getSimpleName()
+                        + " because previous cycle has no " + typeName).withStatus(Status.SKIP);
+                return;
+            }
+            BitSet removedAndModified = calculateRemovedAndModifiedOrdinals(latestOrdinals, previousOrdinals);
+            if (removedAndModified.isEmpty()) {
+                statusBuilder.withMessage("Skipping " + ObjectModificationValidator.class.getSimpleName()
+                        + " because " + typeName + " has no removals or modifications").withStatus(Status.SKIP);
+                return;
+            }
+            HollowObjectTypeReadState objectTypeState = (HollowObjectTypeReadState) typeState;
+            PrimaryKey key = objectTypeState.getSchema().getPrimaryKey();
+            // this is guaranteed to give us items from the most recent cycle, not the last one
+            HollowPrimaryKeyIndex index = new HollowPrimaryKeyIndex(readState.getStateEngine(), key);
+            int fromOrdinal = removedAndModified.nextSetBit(0);
+            while (fromOrdinal != HollowConstants.ORDINAL_NONE) {
+                Object[] candidateKey = index.getRecordKey(fromOrdinal);
+                int matchedOrdinal = index.getMatchingOrdinal(candidateKey);
+                if (matchedOrdinal != HollowConstants.ORDINAL_NONE) {
+                    T fromObject = (T) new GenericHollowObject(objectTypeState, fromOrdinal);
+                    T toObject = (T) new GenericHollowObject(objectTypeState, matchedOrdinal);
+                    if (!filter.apply(fromObject, toObject)) {
+                        throw new ValidationException("Validation failed for key " + Arrays.toString(candidateKey));
+                    }
+                }
+                fromOrdinal = removedAndModified.nextSetBit(fromOrdinal + 1);
+            }
+            statusBuilder.success();
+        } catch (RuntimeException e) {
+            statusBuilder.withMessage(e.getMessage()).fail(e);
+            throw e;
+        }
+    }
+
+	@Override
+	public String toString() {
+        SingleValidationStatus status = statusBuilder.build();
+        return getClass().getSimpleName() + "<typeName=" + typeName + ", status=" + status.getStatus()
+                + ", message=" + status.getMessage() + ", additionalInfo=" + status.getAdditionalInfo()
+                + ", exception=" + status.getThrowable() + ">";
+    }
+
+    private static BitSet calculateRemovedAndModifiedOrdinals(BitSet latestOrdinals, BitSet previousOrdinals) {
+        // make sure we don't modify previousOrdinals or latestOrdinals
+        BitSet removedAndModified = new BitSet();
+        removedAndModified.or(previousOrdinals);
```

Comment:
`return previousOrdinals.clone().andNot(latestOrdinals);`

Reply:
Paul and I look at this together, but it gets messy because 1. clone returns an Object that needs to be cast back to a BitSet, and 2. andNot is a void function, so that needs to be on a separate line from the return anyway.